app/models/odk/duplicate_checker.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module ODK
+  # Checks if there is a duplicate odk xml and any attachments
+  # Works with multiple attachments
+  class DuplicateChecker
+    attr_accessor :files, :user_id
+
+    def initialize(files, user)
+      @files = files
+      @user = user
+    end
+
+    def duplicate?
+      existing_xml_blobs = find_blobs(@files[:xml_submission_file])
+      return false unless dupe_response_and_user?(existing_xml_blobs)
+
+      # If multi part submission, check other attachments
+      other_files = @files.except(:xml_submission_file)
+      return true if other_files.blank?
+
+      other_files.each do |f|
+        return true if dupe_response_and_user?(find_blobs(f[1]))
```

Comment:
i think this will return a false positive if there are 2 images and either of them (but not both) is a duplicate -- you could use `other_files.all? { |f| dupe?(f) }` to check every one instead of any one.

Reply:
in which case, we should have a spec for multiple media attachments :D 

(unless that's difficult for some reason and you can determine that Collect only ever attaches 1 media per submission, which seems plausible but I don't want to rely on it unless we have to)

Reply:
thanks for thinking of this edge case! Given this description from here: https://docs.getodk.org/openrosa-form-submission/#form-submission

It seems that it could be multiple parts.... but not sure if that is how they actually implemented it?

Reply:
i'm gonna do a multiple media attachment spec...i don't trusts it

Reply:
I think this edge case is still present (the loop ends early with "yes it's duplicate" as soon as it finds a file that's duplicated, even if there may be a file later on that's not yet saved) -- can be resolved by switching to `all?` instead of `each`